core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java

```diff
@@ -760,6 +763,58 @@ public void markUsedOutputGuardRailsUnremovable(List<AiServicesMethodBuildItem>
         }
     }
 
+    /**
+     * Because the tools execution uses an imperative API (`String execute(...)`) and uses the caller thread, we need
+     * to anticipate the need to dispatch the invocation on a worker thread.
+     * This is the case for AI service methods that returns `Uni`, `CompletionStage` and `Multi` (stream) and that uses
+     * tools returning `Uni`, `CompletionStage` or that are blocking.
+     * Basically, for "reactive AI service method, the switch is necessary except if all the tools are imperative (return `T`)
+     * and marked explicitly as blocking (using `@Blocking`).
+     *
+     * @param method the AI method
+     * @param tools the tools
+     */
+    public boolean detectAiServiceMethodThanNeedToBeDispatchedOnWorkerThread(
+            MethodInfo method,
+            List<String> associatedTools,
+            List<ToolMethodBuildItem> tools) {
+        boolean reactive = method.returnType().name().equals(DotName.createSimple(Uni.class.getName()))
+                || method.returnType().name().equals(DotName.createSimple(CompletionStage.class.getName()))
+                || method.returnType().name().equals(DotName.createSimple(Multi.class.getName()));
```

Comment:
Minor nitpick: I am pretty sure these `DotName` classes already exist as constants in `DotNames`

Reply:
+1